services/application.js

```diff
@@ -191,11 +191,13 @@ class Application {
 
     async submitApplication(params, context) {
         verifySession(context)
-            .verifyInitialized();
+            .verifyInitialized()
+            .verifyRole([USER.ROLES.SUBMITTER, USER.ROLES.FEDERAL_LEAD])
         const application = await this.getApplicationById(params._id);
         verifyApplication(application)
             .notEmpty()
-            .state([NEW, IN_PROGRESS]);
+            // Submitter or Fed Leads only can submit under certain application status
+            .state(context?.userInfo?.role === USER.ROLES.SUBMITTER ? [NEW, IN_PROGRESS] : [INQUIRED, IN_PROGRESS]);
```

Comment:
I know that with the current code the only possible roles at this line are SUBMITTER and FEDERAL_LEAD but I think we should do a role check anyway to prevent issues resulting from any future changes. If any non-submitters reach this state they will have the same submit permissions as FEDERAL_LEADS

Reply:
I've pushed the change. The empty status will be assigned for invalid roles.